massa-client/src/cmds.rs

```diff
@@ -1069,8 +1257,16 @@ async fn send_operation(
 
 /// TODO: ugly utilities functions
 /// takes a slice of string and makes it into a `Vec<T>`
-pub fn parse_vec<T: std::str::FromStr>(args: &[String]) -> anyhow::Result<Vec<T>, T::Err> {
-    args.iter().map(|x| x.parse::<T>()).collect()
+pub fn parse_vec<T: std::str::FromStr>(args: &[String]) -> anyhow::Result<Vec<T>, Error>
+where
+    T::Err: Display,
```

Comment:
Not sure why we need to add a constraint here. Cannot we just format with {:?} ?

Reply:
`parse_vec` was flagged as UGLY method. Any improvements are welcome. I tried to make it better

Reply:
Not sure to understand... Who flagged this function as ugly? and why?

Reply:
@AurelienFT should knows the history of this method

Reply:
```rust
/// TODO: ugly utilities functions
/// takes a slice of string and makes it into a `Vec<T>`
pub fn parse_vec<T: std::str::FromStr>(args: &[String]) -> anyhow::Result<Vec<T>, Error>
where
    T::Err: Display,
{
    args.iter()
        .map(|x| {
            x.parse::<T>()
                .map_err(|e| anyhow!("failed to parse \"{}\" due to: {}", x, e))
        })
        .collect()
}
```

I need to display which value is causing the error to the user. Feel free if you have a better idea to implement this